lib/features/composer/domain/usecases/send_email_interactor.dart

```diff
@@ -1,39 +1,67 @@
 import 'package:core/core.dart';
 import 'package:dartz/dartz.dart';
 import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
+import 'package:model/extensions/email_extension.dart';
+import 'package:model/mailbox/presentation_mailbox.dart';
+import 'package:tmail_ui_user/features/composer/domain/exceptions/send_email_exception.dart';
 import 'package:tmail_ui_user/features/composer/domain/model/email_request.dart';
 import 'package:tmail_ui_user/features/composer/domain/state/send_email_state.dart';
 import 'package:tmail_ui_user/features/email/domain/repository/email_repository.dart';
+import 'package:tmail_ui_user/features/mailbox/domain/model/create_new_mailbox_request.dart';
 import 'package:tmail_ui_user/features/mailbox/domain/repository/mailbox_repository.dart';
+import 'package:uuid/uuid.dart';
 
 class SendEmailInteractor {
   final EmailRepository _emailRepository;
   final MailboxRepository _mailboxRepository;
+  final Uuid _uuid;
 
-  SendEmailInteractor(this._emailRepository, this._mailboxRepository);
+  SendEmailInteractor(this._emailRepository, this._mailboxRepository, this._uuid);
 
   Stream<Either<Failure, Success>> execute(AccountId accountId, EmailRequest emailRequest) async* {
     try {
       yield Right<Failure, Success>(SendingEmailState());
+      final newOutboxMailboxId = emailRequest.outboxMailboxId ?? await _autoGenerateMissingOutboxMailbox(accountId);
```

Comment:
we can not `create mailbox -> then send email` in the same Jmap request?

Reply:
I tried and it's too hard to take a `property` like `id` in the returned result and reassign it to an object created from the client? If you have a solution for this please suggest me. Thanks